src/GeometryCast.php

```diff
@@ -76,23 +75,16 @@ public function set($model, string $key, $value, array $attributes): ?Expression
         return $value->toSqlExpression($model->getConnection());
     }
 
-    private function extractWktFromExpression(ExpressionContract $expression, Connection $connection): string
-    {
-        $grammar = $connection->getQueryGrammar();
-        $expressionValue = $expression->getValue($grammar);
-
-        preg_match('/ST_GeomFromText\(\'(.+)\', .+(, .+)?\)/', (string) $expressionValue, $match);
-
-        return $match[1];
-    }
-
-    private function extractSridFromExpression(ExpressionContract $expression, Connection $connection): int
+    /**
+     * @return array{string, int, string}
+     */
+    private function extractFromExpression(ExpressionContract $expression, Connection $connection): array
     {
         $grammar = $connection->getQueryGrammar();
         $expressionValue = $expression->getValue($grammar);
 
-        preg_match('/ST_GeomFromText\(\'.+\', (.+)(, .+)?\)/', (string) $expressionValue, $match);
+        preg_match("/ST_GeomFromText\(\s*'([^']+)'\s*(?:,\s*(\d+))?\s*(?:,\s*'([^']+)')?\s*\)/", (string) $expressionValue, $matches);
 
-        return (int) $match[1];
+        return [$matches[1], (int) ($matches[2] ?? 0), $matches[3] ?? ''];
```

Comment:
I think this would increase readability.

```suggestion
        return ['wkt' => $matches[1], 'srid' => (int) ($matches[2] ?? 0), 'options' => $matches[3] ?? ''];
```

Reply:
BTW, I don't think returning `options` is necessary as we don't use it today. I would appreciate if you could remove it.